### PR TITLE
DBZ-7765: Use different query for validating log position for SQLServer

### PR DESCRIPTION
**Pull request by mfvitale**
closes: https://issues.redhat.com/browse/DBZ-7765

### CODE REVIEW

File: debezium-core/src/main/java/io/debezium/config/CommonConnectorConfig.java [hunk 1]
```diff
@@ -1043,6 +1044,16 @@ public static SnapshotQueryMode parse(String value, String defaultValue) {
             .withDescription(
                     "When 'snapshot.mode' is set as configuration_based, this setting permits to specify whenever the data should be snapshotted or not in case of error.");
 
+    public static final Field LOG_POSITION_CHECK_ENABLED = Field.create("log.position.check.enable")
```
Comment (jpechane) [inline debezium-core/src/main/java/io/debezium/config/CommonConnectorConfig.java:1047]:
```suggestion
    public static final Field LOG_POSITION_CHECK_ENABLED = Field.createInternal("log.position.check.enable")
```

let's keep it undocumented at the start till we evaluate if it is something necessary

**Comment by jpechane** [comment]
@mfvitale Applied, thanks

**Comment by jpechane** [comment]
@mfvitale Could you please prepare backport to 2.6? Thanks

**Comment by mfvitale** [comment]
@jpechane Done.

**Pull request merged**